sympy/physics/vector/dyadic.py

```diff
@@ -302,32 +302,37 @@ def __rxor__(self, other):
             ol += v[0] * ((other ^ v[1]) | v[2])
         return ol
 
-    def __str__(self, printer=None):
+    def _sympystr(self, printer):
         """Printing method. """
         ar = self.args  # just to shorten things
         if len(ar) == 0:
-            return str(0)
+            return printer._print(0)
+
+        # Ignore parent printer class and settings and use our own.
+        # TODO: Remove this, it's only here to preserve old behavior
+        printer = VectorStrPrinter()
```

Comment:
This pattern is used in lots of places throughout `physics.vector`. It clearly needs fixing, but I don't want to change behavior in the same PR as a cleanup.